dashboard/lib/build_dashboard_page.dart

```diff
@@ -36,21 +37,51 @@ class BuildDashboardPageState extends State<BuildDashboardPage> {
   TaskGridFilter _filter;
   TaskGridFilter _settingsBasis;
 
+  /// Current Flutter repository to view.
+  String repo;
+
+  /// Git branch in [repo] to view.
+  ///
+  /// The frontend will update default branches based on [defaultBranches]. This enables users to easily switch from
+  /// master on one repo, to main for a different repo.
+  String branch;
+
   @override
   void initState() {
     super.initState();
+    if (widget.queryParameters != null) {
+      repo = widget.queryParameters['repo'] ?? 'flutter';
+      branch = widget.queryParameters['branch'] ?? 'master';
```

Comment:
nit: you could remove the "?? 'flutter'" and "?? 'master'" here since that's duplicated in line 56/57 below.